### ISSUE TITLE
torch.nn.TransformerEncoder.forward() references attn_mask, which is not an argument.

### ISSUE THREAD

**Issue by delkind-dnsf** [pytorch/pytorch#103431]
### 📚 The doc issue

The docs for torch.nn.TransformerEncoder.forward() includes this description of `is_causal`. 

> is_causal ([Optional](https://docs.python.org/3/library/typing.html#typing.Optional)[[bool](https://docs.python.org/3/library/functions.html#bool)]) – If specified, applies a causal mask as mask (optional) and ignores attn_mask for computing scaled dot product attention. Default: False.

However, `attn_mask` is not an argument in this method. There are two arguments that pertain to masks:
1. `mask`
2. `src_key_padding_mask`

### Suggest a potential alternative/fix

I think it would be clearer to refer to one of these arguments by name; if `is_causal` does pertain to either mask, then the documentation should explain what it does pertain to. Because of the inconsistent naming, it's not obvious whether either or both of these mask arguments are related to `is_causal`. 

I *believe* the intention is that `is_causal` is a convenient substitute for `mask` in certain usages. Because a large number of transformer usages are about next-token prediction tasks, the `is_causal` flag lets PyTorch generate `mask` on behalf of users so that the model only attends to previous tokens.

If I'm correct, the fix could be as simple as

> is_causal ([Optional](https://docs.python.org/3/library/typing.html#typing.Optional)[[bool](https://docs.python.org/3/library/functions.html#bool)]) – If specified, applies a causal mask as `mask` (optional) and ignores `mask` for computing scaled dot product attention. Default: False.

But if I'm not correct, then the docs should be revised for clarity. What is `attn_mask` and how does it relate to `torch.nn.TransformerEncoder.forward()`?

**Comment by mikaylagawarecki** [comment]
Hey, this docstring was recently updated and is reflected in the unstable docs (https://pytorch.org/docs/main/generated/torch.nn.TransformerEncoder.html#transformerencoder) it will be reflected in the next release. 

Alternatively, you could toggle the dropdown here to see these docs 
<img width="431" alt="Screenshot 2023-06-13 at 1 05 18 PM" src="https://github.com/pytorch/pytorch/assets/35276741/5116e3a6-f96e-407f-926a-582c8ed697a8">

Closing this issue as this has ben fixed. Feel free to reopen if you have further questions
